### PR TITLE
Add Anomaly Task

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algo/anomaly/openvino_model.py [hunk 1]
```diff
@@ -105,10 +105,11 @@ def __init__(
         model_api_configuration: dict[str, Any] | None = None,
         metric: MetricCallable = NullMetricCallable,  # Metrics is computed using Anomalib's metric
         task: Literal[
-            OTXTaskType.ANOMALY_CLASSIFICATION,
+            OTXTaskType.ANOMALY,
+            OTXTaskType.ACTION_CLASSIFICATION,
```
Comment (samet-akcay) [inline src/otx/algo/anomaly/openvino_model.py]: ?
```suggestion
            OTXTaskType.ANOMALY_CLASSIFICATION,
```

File: src/otx/core/types/task.py [hunk 2]
```diff
@@ -15,6 +15,8 @@ class OTXTaskType(str, Enum):
     ACTION_CLASSIFICATION = "ACTION_CLASSIFICATION"
 
     # Anomaly Detection
+    # TODO(ashwinvaidya17): Remove rest after greenlight from backend.
+    ANOMALY = "ANOMALY"
```
Comment (samet-akcay) [inline src/otx/core/types/task.py]:
```suggestion
    ANOMALY = "ANOMALY"
    # TODO(ashwinvaidya17): Remove rest after greenlight from backend.    
```

**Comment by chuneuny-emily** [comment]
Please do not merge this PR into releases/2.2.0 branch until OTX 2.2.0 standalone is released. Since this feature is targeting Geti 2.5, it should be merged in OTX2.2.x.

**Comment by jihyeonyi** [comment]
> Please do not merge this PR into releases/2.2.0 branch until OTX 2.2.0 standalone is released. Since this feature is targeting Geti 2.5, it should be merged in OTX2.2.x.

Hi, I think we need to discuss about the target, because Mark said this should be merged in OTX 2.3.

**Comment by kprokofi** [comment]
@ashwinvaidya17, do we have urgency to merge this PR to OTX2.2 release? Since Code Freeze passed, we shouldn't merge anything. 
If it is not urgent for Geti integration, we need to target it to OTX2.3 release (develop)

**Comment by ashwinvaidya17** [comment]
> @ashwinvaidya17, do we have urgency to merge this PR to OTX2.2 release? Since Code Freeze passed, we shouldn't merge anything. If it is not urgent for Geti integration, we need to target it to OTX2.3 release (develop)

I don't think there is urgency. From what I was told, the 1.6.x one is more important

**Comment by ashwinvaidya17** [comment]
Closing as https://github.com/openvinotoolkit/training_extensions/pull/3910 adds changes to develop

**Pull request closed without merging**